### PR TITLE
Stats: Chart refresh header component update

### PR DESCRIPTION
**Pull request by annacmc**
Related to https://github.com/Automattic/red-team/issues/249

## Proposed Changes

* This PR adds a new header component for the chart refresh. 
* **NOTE: there is NO styling yet**
* All CSS styling in next PR
* Adding the period drop down selector is in the following PR

## Why are these changes being made?

* Part of the chart refresh for the new date filtering traffic page. The changes introduce a new ChartHeader component, which is consistent with React's component-based architecture. By separating the header logic into its own component, the code adheres to the common React pattern of modular and reusable components. This approach enhances maintainability and readability. 

## Testing Instructions

* Open Calypso Live Branch
* Ensure that the chart on the traffic page still looks exactly the same as before. 
* Apply feature flag `stats/new-date-filtering`
* Ensure that the chart on the traffic page is now in its own header component, and has an (unstyled) label for the currently selected (active) tab to the left. 

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-chart-tabs/chart-header.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import PropTypes from 'prop-types';
```
Comment (kangzj) [inline client/my-sites/stats/stats-chart-tabs/chart-header.jsx:1]: We should probably start new components in TypeScript, where PropTypes wouldn't be necessary anymore.

**Review by kangzj — APPROVED** [review]
Works as described. Good work 👍 

I wonder if we can apply what we learned from the composition method whether we could treat the header as a slot, and we fill in the right header for current and apply the new header later. I'm curious to know the next move to keep the current looks + implement the new look 🤔

File: client/my-sites/stats/stats-chart-tabs/chart-header.jsx (hunk 1 above)
Comment (annacmc) [reply client/my-sites/stats/stats-chart-tabs/chart-header.jsx:1]: Yeah totally. Adding this to the followup PR which will also address your excellent suggestion above. I've been pondering how to use this as a slot so will try figure it out 😄

Comment (annacmc) [reply client/my-sites/stats/stats-chart-tabs/chart-header.jsx:1]: issue created to followup here: https://github.com/Automattic/red-team/issues/254

**Pull request merged**